### PR TITLE
Add basic permissions to the users api

### LINKED ISSUE #3684: Implement Proper permissions for the users api
- GET /users (Restricted to super-admin / (admin with relevant access) )
- POST /users (Public)
- All methods /users/<id> (Restricted to super-admin / (admin with relevant access) )

Some more endpoints need to be defined for the users api. Another issue is being opened for them

### PR DESCRIPTION
**Pull request by shubham-padia**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Add basic permissions to the users api

#### Changes proposed in this pull request:

- Add basic permissions to the users api
- Add permission decorators to `api/helpers/permissions.py`
- Add ErrorResponse class for jsonapi compliant response handling
- GET /users (Restricted to super-admin / (admin with relevant access) )
- POST /users (Public)
- All methods /users/<id> (User can access his/her own id ) / ( Restricted to super-admin / admin ) 

Fixes #3684

### CODE REVIEW

File: app/api/helpers/permissions.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+from functools import wraps
+from flask import current_app as app
+from flask_jwt import _jwt_required, current_identity
+from app.api.helpers.errors import ForbiddenError
+
+
+def decdec(inner_dec):
```
Comment (SaptakS) [inline app/api/helpers/permissions.py]: I am not very sure but the variable names seem very confusing. It should be more intuitive I think.

Comment (niranjan94) [inline app/api/helpers/permissions.py]: Agreed with @SaptakS . @shubham-padia please change the name

Comment (shubham-padia) [inline app/api/helpers/permissions.py]: @SaptakS @niranjan94 done.

**Comment by niranjan94** [comment]
@shubham-padia can you give a sample of the error response ?

**Comment by shubham-padia** [comment]
```
{
  "errors": [
    {
      "status": 403,
      "source": {
        "source": ""
      },
      "detail": "You need admin access",
      "title": "Access Forbidden"
    }
  ],
  "jsonapi": {
    "version": "1.0"
  }
}
```

File: app/api/helpers/permissions.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    """
+    Decorator function for things allowed to admins and super admins.
+    :param f:
+    :return:
+    """
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        user = current_identity
+        if not user.is_admin and not user.is_super_admin:
+            return ForbiddenError({'source': ''}, 'You need admin access').respond()
```
Comment (niranjan94) [inline app/api/helpers/permissions.py]: Change text to `Admin access required`

File: app/api/helpers/permissions.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    Decorator function for things allowed exclusively to super admin.
+    Do not use this if the resource is also accessible by a normal admin, use the is_admin decorator instead.
+    :param f:
+    :return:
+    """
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        user = current_identity
+        if not user.is_super_admin:
+            return ForbiddenError({'source': ''}, 'You need super admin access').respond()
```
Comment (niranjan94) [inline app/api/helpers/permissions.py]: Change `Super admin access is required`

Comment (shubham-padia) [inline app/api/helpers/permissions.py]: @niranjan94 changes done.

**Pull request merged**